### PR TITLE
Upgrade psalm

### PR DESCRIPTION
**Pull request by stof**

### CODE REVIEW

File: .github/workflows/static.yml [hunk 1]
```diff
@@ -105,7 +105,7 @@ jobs:
           composer update --no-interaction --prefer-dist --optimize-autoloader
 
       - name: Psalm
-        uses: docker://vimeo/psalm-github-actions:5.1.0
+        uses: docker://ghcr.io/psalm/psalm-github-actions:5.11.0
```
Comment (GrahamCampbell) [inline .github/workflows/static.yml:108]: can we install the psalm phar, similar to the way we do phpstan?

```
"psalm/phar": "5.11.0"
```

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Review by jderusse — COMMENTED** [review]
Is there a reason for not using composer like we did for phpstan and BC-break-checker?

**Comment by stof** [comment]
@jderusse I kept it closer to the way it was done right now. But I'm strongly in favor of using composer

**Comment by jderusse** [comment]
> @jderusse I kept it closer to the way it was done right now. But I'm strongly in favor of using composer

I'm also in favor of migrating to composer: Docker images are not always maintained, and also prevent contributor from running tools on their local environments

**Comment by stof** [comment]
Well, running Psalm locally is still a pain because it requires installing dependencies with an old PHP version as target (due to the fact that Psalm defaults to analyzing the code with the lowest bound of the composer requirement for PHP but also insists that the code scanned in the vendor folder can be analyzed with this PHP version)

**Comment by stof** [comment]
I have no idea why the CI fails. Running the same Psalm version locally does not report the error.

**Comment by GrahamCampbell** [comment]
Possibly a bad local psalm cache. psalm is pretty bad at getting itself in a situation with an incorrect cache if things are moving around while it is running.

**Comment by stof** [comment]
I'm getting this difference even with the psalm cache deleted and the `--no-cache` option

**Comment by GrahamCampbell** [comment]
Hmm. 🤷

**Comment by stof** [comment]
I still don't know why my local setup was not detecting this error. But as it is a legitimate one, I fixed it in the PR.

**Review by jderusse — APPROVED** [review]

**Pull request merged**